contrib_bots/lib/sched.py

```diff
@@ -0,0 +1,186 @@
+# pip install schedule is necessary
+import schedule
+import time
+import simplejson as json
+import threading
+from random import randint
+
+class SchedHandler(object):
+    '''
+    This bot allows you to schedule daily alerts. It
+    looks for messages starting with '@every' or '@cancel' or '@recover'.
+
+    In this example, we write alerts at certain time intervals
+    in a new stream called "alert,".
+    '''
+
+    def usage(self):
+        def run_threaded(job_func):
+            job_thread = threading.Thread(target=job_func)
+            job_thread.start()
+
+        def event_handler():
+            time.sleep(5)
+            schedule.run_pending()
+
+        schedule.every(1).seconds.do(run_threaded, event_handler)
+        time.sleep(2)
+        schedule.run_pending()
+        return '''
+            Users should preface messages with "@every" or '@cancel' or '@recover'.
+
+            Before running this, make sure to create a stream
+            called "alert" that your API user can send to.
+            '''
+
+    def triage_message(self, message):
+        # return True if we want to (possibly) response to this message
+
+        original_content = message['content']
+
+        is_link = (original_content.startswith('@every') or
+                   original_content.startswith('@cancel') or
+                   original_content.startswith('@recover'))
+
+        return is_link
+
+    def handle_message(self, message, client, state_handler):
+        # For every task we want to schedule we define a job that have a random unique id if we want to cancel the task.
+        def job(randomint):
+            original_content = message['content']
+            a = original_content.split()[3:]
+            b = " ".join(a)
+            try:
+                if b.split()[0] == "at":
+                    b = b.split()[2:]
+                    b = " ".join(b)
+            except IndexError:
+                pass
+            events = dict(
+                type='stream',
+                to='alert',
+                subject="All alerts",
+                content=b+" "+randomint,
+            )
+            client.send_message(events)
+
+        original_content = message['content']
+        original_sender = message["sender_email"]
+        a = str(randint(1, 1000000))
+        key = dict(
+            type='stream',
+            to='alert',
+            subject="All alerts",
+            content=original_content+" "+a,
+            )
+
+        # Function needed to verify if hour supplied meets syntax.
+        def isTimeFormat(input):
```

Comment:
By convention Python uses snake case for naming functions:

https://www.python.org/dev/peps/pep-0008/#function-names

Also `input` is a builtin function and you are overwriting it here